### ISSUE TITLE
Deprecate helper-npm-install-peer-deps

### ISSUE THREAD

**Issue by simonplend** [Financial-Times/next-ci-shared-helpers#7]
- [ ] Generate build state artifact that lists missing peer dependencies ([helper-generate-build-state-artifacts](https://github.com/Financial-Times/next-ci-shared-helpers/blob/master/helper-generate-build-state-artifacts))
- [ ] Update [helper-npm-install-peer-deps](https://github.com/Financial-Times/next-ci-shared-helpers/blob/master/helper-npm-install-peer-deps) so it no longer installs missing peer dependencies, but refers to the new build state artifact
- [ ] Update config templates in [n-circle2-cli](https://github.com/Financial-Times/n-circle2-cli) so they no longer call the `helper-npm-install-peer-deps`

**Comment by adgad** [comment]
This sounds good to me.

Am I right in thinking that most projects currently aren't installing peer deps from this hack (i.e. is `ajv` actually still a thing?). If it is, perhaps we should fix the common ones in n-ui or something, so for most apps it's seamless?

**Comment by simonplend** [comment]
The peer deps helper was installing any missing peer deps and then npm update was run in the next step, which was removing the peer deps that were just installed as they were extraneous. I fixed that in our CircleCI config templates here so that these extraneous packages are left alone: https://github.com/Financial-Times/n-circle2-cli/pull/11. The work I'm proposing above is a follow on from this.

Regarding `ajv` I saw the conflict yesterday on `next-article` because it was depending on the latest major version of `ajv` (presumably as an attempt to fix a peer dep issue in the past), but `n-gage` depends on old major versions of `eslintrc` and `stylelint`, which creates the following issue...

```

**Comment by adgad** [comment]
Okay that sounds good. I guess I was mainly checking that we weren't going to suddenly break hundreds of builds. If it's just the odd couple that have to fix their deps that's fine.

**Comment by simonplend** [comment]
Yeah, I'm pretty sure this *won't* break hundreds of builds, but that's why I want to take it one step at a time to see how things look before fully deprecating the `helper-npm-install-peer-deps` :smile:
